src/config.rs

```diff
@@ -120,15 +141,55 @@ pub struct Config {
     #[structopt(long, parse(try_from_str = parse_millis), value_name = "MILLIS")]
     pub upnp_lease_duration: Option<Duration>,
 
-    /// How long to retry establishing connections and sending messages.
+    /// The initial retry interval.
+    ///
+    /// This is the first delay before a retry, for establishing connections and sending messages.
+    /// The subsequent delay will be decided by the `retry_delay_multiplier`.
+    ///
+    /// If unspecified, this will default to [`DEFAULT_INITIAL_RETRY_INTERVAL`].
+    #[serde(default)]
+    #[structopt(long, parse(try_from_str = parse_millis), value_name = "MILLIS")]
+    pub initial_retry_interval: Option<Duration>,
```

Comment:
We could go for `pub retry: Option<RetryConfig>`. It's not much worse for callers, since they can do:

```rust
Config {
    retry: RetryConfig {
        field_i_care_about: Duration::from_secs(123),
        ..Default::default()
    }
}
```

But it makes quite a lot of stuff simpler, deduplicates docs, etc.

Reply:
Yes I thought of that as well, but then there was some structop thing and I just skipped it for now.
We can do a follow up with that.

Btw, ` ..Default::default()` does that insert the defaults of the struct, or the defaults of the types of the fields?

Reply:
My only hesitance is that follow up would mean more breaking change and API churn (e.g. introducing new APIs here, starting to use them, then replacing them all soon after). Perhaps we could remove `StructOpt` instead? It seems quite misplaced here, but I'm not sure what could be using it (`safe_network` at least [seems safe](https://github.com/maidsafe/safe_network/blob/2b29dbbcb6b9ef5c1a5b3758511dd23aa32a7c60/src/node/config_handler.rs#L129-L131)).

I guess is removing `StructOpt` looks non-trivial, we should deal with it as a follow-up, but if it's just a case of dropping it because we don't use it, I think that would be best.

Reply:
`Structop` dropped now.